addons/crm_claim/models/crm_claim.py

```diff
@@ -0,0 +1,136 @@
+
+    def _get_default_stage_id(self):
+        """ Gives default stage_id """
+        team_id = self.env['crm.team']._get_default_team_id()
+        return self.stage_find([], team_id, [('sequence', '=', '1')])
+
+    name = fields.Char(string='Claim Subject', required=True)
+    active = fields.Boolean(default=True)
+    action_next = fields.Char(string='Next Action')
+    date_action_next = fields.Datetime(string='Next Action Date')
+    description = fields.Text()
+    resolution = fields.Text()
+    create_date = fields.Datetime(string='Creation Date', readonly=True)
+    write_date = fields.Datetime(string='Update Date', readonly=True)
+    date_deadline = fields.Date(string='Deadline')
+    date_closed = fields.Datetime(string='Closed', readonly=True)
+    date = fields.Datetime(string='Claim Date', index=True, default=fields.Datetime.now)
+    ref = fields.Reference(selection=lambda self: referencable_models(self, self.env.cr, self.env.uid, self.env.context), string='Reference')
+    categ_id = fields.Many2one('crm.claim.category', string='Category')
+    priority = fields.Selection([('0', 'Low'), ('1', 'Normal'), ('2', 'High')], default='1')
+    type_action = fields.Selection([('correction', 'Corrective Action'), ('prevention', 'Preventive Action')], string='Action Type')
+    user_id = fields.Many2one('res.users', string='Responsible', track_visibility='always', default=lambda self: self.env.uid)
+    user_fault = fields.Char(string='Trouble Responsible')
+    team_id = fields.Many2one(
+        'crm.team', 'Sales Team', index=True,
+        help="Responsible sales team."
+        " Define Responsible user and Email account for"
+        " mail gateway.", default=lambda self: self.env['crm.team']._get_default_team_id(), oldname='section_id')
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.user.company_id)
+    partner_id = fields.Many2one('res.partner', string='Partner')
+    email_cc = fields.Text(string='Watchers Emails', size=252, help="These email addresses will be added to the CC field of all inbound and outbound emails for this record before being sent. Separate multiple email addresses with a comma")
+    email_from = fields.Char(string='Email', size=128, help="Destination email for email gateway.")
+    partner_phone = fields.Char(string='Phone')
+    stage_id = fields.Many2one('crm.claim.stage', string='Stage', track_visibility='onchange', domain="['|', ('team_ids', '=', team_id), ('case_default', '=', True)]", default=_get_default_stage_id)
+    cause = fields.Text(string='Root Cause')
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        """This function returns value of partner address based on partner
+           :param email: ignored
+        """
+        self.email_from = self.partner_id.email
+        self.partner_phone = self.partner_id.phone
+
+    def stage_find(self, cases, team_id=False, domain=[], order='sequence'):
```

Comment:
@rap-odoo i think @hmo-odoo right because when this method calling from old-api , code may be huhhhh.... 
![giphy](https://cloud.githubusercontent.com/assets/14940179/12152298/16d07924-b471-11e5-8f03-60c5df25c26c.gif)